src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferDictionaryContainsMethods.cs

```diff
@@ -0,0 +1,208 @@
+
+                if (property.Name == KeysPropertyName)
+                {
+                    if (!containsParameterType.Equals(keyType, SymbolEqualityComparer.Default))
+                        return;
+
+                    //  Now we search for an accessible ContainsKey method that returns boolean and accepts a single
+                    //  parameter of the type that was substituted for TKey.
+
+                    IMethodSymbol? containsKeyMethod = property.ContainingType.GetMembers(ContainsKeyMethodName)
+                        .OfType<IMethodSymbol>()
+                        .WhereAsArray(x => x.ReturnType.SpecialType == SpecialType.System_Boolean &&
+                            x.IsPublic() &&
+                            x.Parameters.Length == 1 &&
+                            x.Parameters[0].Type.Equals(keyType, SymbolEqualityComparer.Default))
+                        .FirstOrDefault();
+
+                    if (containsKeyMethod is null)
+                        return;
+
+                    var diagnostic = invocation.CreateDiagnostic(ContainsKeyRule, property.ContainingType.Name);
+                    context.ReportDiagnostic(diagnostic);
+                }
+                else if (property.Name == ValuesPropertyName)
+                {
+                    if (!containsParameterType.Equals(valueType, SymbolEqualityComparer.Default))
+                        return;
+
+                    //  Now we search for an accessible ContainsValue method that returns boolean and accepts a single
+                    //  parameter of the type that was substituted for TValue.
+
+                    IMethodSymbol? containsValueMethod = property.ContainingType.GetMembers(ContainsValueMethodName)
+                        .OfType<IMethodSymbol>()
+                        .WhereAsArray(x => x.ReturnType.SpecialType == SpecialType.System_Boolean &&
+                            x.IsPublic() &&
+                            x.Parameters.Length == 1 &&
+                            x.Parameters[0].Type.Equals(valueType, SymbolEqualityComparer.Default))
+                        .FirstOrDefault();
+
+                    if (containsValueMethod is null)
+                        return;
+
+                    var diagnostic = invocation.CreateDiagnostic(ContainsValueRule, property.ContainingType.Name);
+                    context.ReportDiagnostic(diagnostic);
+                }
+            }
+
+            static bool TryGetPropertyReferenceOperation(IInvocationOperation containsInvocation, [NotNullWhen(true)] out IPropertySymbol? property)
+            {
+                IMethodSymbol method = containsInvocation.TargetMethod;
+                IOperation? receiver = null;
+
+                //  C# and VB handle extension methods differently. In C#, the "this" argument is included in the argument list. In VB, it's not.
+                if (method.IsExtensionMethod && method.Language == LanguageNames.CSharp && method.Parameters.Length == 2)
```

Comment:
Please remove language name checks and instead introduce abstract methods with language specific handling in the language specific layers.

Reply:
Fixed.